packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -1036,53 +989,213 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
   }
 }
 
+typedef _AssignSemantics = void Function(RenderSemanticsGestureHandler);
+
 class _GestureSemantics extends SingleChildRenderObjectWidget {
   const _GestureSemantics({
     Key key,
     Widget child,
-    this.owner,
-  }) : super(key: key, child: child);
+    @required this.owner,
+    @required this.assignSemantics,
+  }) : assert(owner != null),
+       assert(assignSemantics != null),
+       super(key: key, child: child);
 
   final RawGestureDetectorState owner;
+  final _AssignSemantics assignSemantics;
 
   @override
   RenderSemanticsGestureHandler createRenderObject(BuildContext context) {
-    return RenderSemanticsGestureHandler(
-      onTap: _onTapHandler,
-      onLongPress: _onLongPressHandler,
-      onHorizontalDragUpdate: _onHorizontalDragUpdateHandler,
-      onVerticalDragUpdate: _onVerticalDragUpdateHandler,
-    );
-  }
-
-  void _updateHandlers(RenderSemanticsGestureHandler renderObject) {
-    renderObject
-      ..onTap = _onTapHandler
-      ..onLongPress = _onLongPressHandler
-      ..onHorizontalDragUpdate = _onHorizontalDragUpdateHandler
-      ..onVerticalDragUpdate = _onVerticalDragUpdateHandler;
+    final RenderSemanticsGestureHandler renderObject = RenderSemanticsGestureHandler();
+    assignSemantics(renderObject);
+    return renderObject;
   }
 
   @override
   void updateRenderObject(BuildContext context, RenderSemanticsGestureHandler renderObject) {
-    _updateHandlers(renderObject);
+    assignSemantics(renderObject);
   }
+}
+
+typedef GetRecognizerHandler = GestureRecognizer Function(Type);
```

Comment:
I'm unsure if I should use a function here or a unmodifiable map.
- Pros of unmodifiable map: One less public typedef, and easier to understand
- Cons of unmodifiable map: Slower because we need to copy the map

Reply:
It is no longer relevant under the new design.